### PR TITLE
FPS-115: GCS Empty Dir Fix

### PR DESCRIPTION
**Pull request by bjohnson-fdx**
Changes:
* Force trailing "/" for GCS virtual directories / folder prefixes 
(cherry picked from commit [`93dfb03`](https://github.com/drakkan/sftpgo/commit/93dfb03eafdd024b7237e9f225b01ac22a1d5532))

---

> [!NOTE]
> **Medium Risk**
> Touches core GCS VFS path semantics (stat/rename/remove/mkdir/readdir), so regressions could impact directory operations and client compatibility, though changes are localized and covered by new tests.
> 
> **Overview**
> Fixes GCS virtual directory edge cases by **consistently normalizing directory object names with a trailing `/`** across `Mkdir`, `Remove`, and `Rename`, while still supporting legacy directory placeholders created without the slash.
> 
> Refactors `Stat` to use a new `getObjectStat` helper that returns both `FileInfo` and the *real* object key (handling the trailing-`/` compat lookup), and updates `ReadDir` to ignore “good” empty directory placeholder objects while still listing “bad” empty dirs needed for some clients (e.g., FileZilla). Adds targeted unit tests covering these empty-dir listing scenarios and placeholder detection. Also updates `.gitignore` to ignore `NOTES.md`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 398afeb390f8ea0723f2b3b2b90bc53eda4443b4. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: vfs/gcsfs_test.go [hunk 1]
```diff
@@ -228,6 +228,126 @@ func (Suite *GCSFsSuite) TestReadDir_IsDir() {
...
+		})
+
+	results, err := Suite.Fs.ReadDir(prefix)
+	Suite.NoError(err)
+	Suite.Len(results, 1)
+
+	// NOTE: this listing MUST be present for FileZilla to display empty "bad" dirs with no trailing `/` in prefix
+	Suite.Equal("bad-empty", results[0].Name())
+	Suite.Equal(int64(0), results[0].Size())
+	Suite.True(results[0].IsDir())
```
[the remark this answers is not in the document]
Comment (bjohnson-fdx) [reply vfs/gcsfs_test.go:318]: see comment on line 311

File: vfs/gcsfs_test.go [hunk 2]
```diff
@@ -228,6 +228,126 @@ func (Suite *GCSFsSuite) TestReadDir_IsDir() {
 	Suite.True(gock.IsDone(), "pending mocks: %s", printPendingMocks())
 }
 
+func (Suite *GCSFsSuite) TestReadDir_NoTrailingSlash_GoodEmptyDir() {
+	defer gock.Off()
+	prefix := "users/test1/good-empty"
+
+	gock.New(testBaseURL).
+		Get("/b/bucket1/o").
+		MatchParam("prefix", prefix).
```
[the remark this answers is not in the document]
Comment (bjohnson-fdx) [reply vfs/gcsfs_test.go:237]: These scenarios were built around actual request/responses from GCS

**Review by kreulenk — APPROVED** [review]

**Review by mmcgeefeedo — APPROVED** [review]
